### PR TITLE
Updated gesture recognition demo

### PR DESCRIPTION
**Pull request by evgeny-izutov**
Updates:
* Renames ASL Recognition demo to Gesture Recognition demo
* Adds `--action_threshold` parameter to redefine default threshold value

### CODE REVIEW

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
please add your demo to the index at [README.md](https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/README.md)

**Comment by evgeny-izutov** [comment]
@vladimir-dudnik The demo is added to README.md

**Review by vladimir-dudnik — APPROVED** [review]

File: demos/python_demos/gesture_recognition_demo/README.md [hunk 1]
```diff
@@ -59,6 +63,7 @@ Options:
   --no_show             Optional. Do not visualize inference results.
   -u UTILIZATION_MONITORS, --utilization_monitors UTILIZATION_MONITORS
                         Optional. List of monitors to show initially.
+
```
Comment (Wovchena) [inline demos/python_demos/gesture_recognition_demo/README.md:66]:
```suggestion
```
Why? :)

File: demos/python_demos/gesture_recognition_demo/README.md [hunk 2]
```diff
@@ -67,26 +72,26 @@ To run the demo, you can use public or pre-trained models. To download the pre-t
 
 > **NOTE**: Before running the demo with a trained model, make sure the model is converted to the Inference Engine format (`*.xml` + `*.bin`) using the [Model Optimizer tool](https://docs.openvinotoolkit.org/latest/_docs_MO_DG_Deep_Learning_Model_Optimizer_DevGuide.html).
 
-To run the demo, please provide paths to the ASL recognition and person detection models in the IR format, to a file with class names, and to an input video:
+To run the demo, please provide paths to the gesture recognition and person detection models in the IR format, to a file with class names, and to an input video:
 ```bash
-python asl_recognition_demo.py \
+python gesture_recognition_demo.py \
 -m_a /home/user/asl-recognition-0003.xml \
```
Comment (Wovchena) [inline demos/python_demos/gesture_recognition_demo/README.md:78]: 0003 model was replaced with 0004. Could you update it while you are here?

File: demos/python_demos/gesture_recognition_demo/gesture_recognition_demo.py [hunk 3]
```diff
@@ -66,11 +65,14 @@ def build_argparser():
...
     args.add_argument('-c', '--class_map',
-                      help='Required. Path to a file with ASL classes.',
+                      help='Required. Path to a file with gesture classes.',
                       required=True, type=str)
     args.add_argument('-s', '--samples_dir',
                       help='Optional. Path to a directory with video samples of gestures.',
                       default=None, type=str)
+    args.add_argument('-t', '--action_threshold',
+                      help='Optional. Threshold for the predicted score of an action.',
+                      default=None, type=float)
```
Comment (Wovchena) [inline demos/python_demos/gesture_recognition_demo/gesture_recognition_demo.py:75]:
```suggestion
                      default=0.8, type=float)
```
So you can remove `ACTION_SCORE_THRESHOLD`.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: demos/python_demos/gesture_recognition_demo/README.md (hunk 1 above)
Comment (evgeny-izutov) [reply demos/python_demos/gesture_recognition_demo/README.md]: Fixed

File: demos/python_demos/gesture_recognition_demo/README.md (hunk 2 above)
Comment (evgeny-izutov) [reply demos/python_demos/gesture_recognition_demo/README.md]: Renamed

File: demos/python_demos/gesture_recognition_demo/gesture_recognition_demo.py (hunk 3 above)
Comment (evgeny-izutov) [reply demos/python_demos/gesture_recognition_demo/gesture_recognition_demo.py]: Done

**Review by Wovchena — APPROVED** [review]

**Review by eizamaliev — APPROVED** [review]

File: demos/README.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ The Open Model Zoo includes the following demos:
 
 - [3D Human Pose Estimation Python* Demo](./python_demos/human_pose_estimation_3d_demo/README.md) - 3D human pose estimation demo.
 - [Action Recognition Python* Demo](./python_demos/action_recognition/README.md) - Demo application for Action Recognition algorithm, which classifies actions that are being performed on input video.
+- [Gesture Recognition Python* Demo](./python_demos/gesture_recognition_demo/README.md) - Demo application for Gesture Recognition algorithm (e.g. American Sign Language gestures), which classifies gesture actions that are being performed on input video.
```
Comment (IRDonch) [inline demos/README.md:9]: Please move this item so that alphabetical order is preserved.

Comment (evgeny-izutov) [reply demos/README.md]: Done.

**Pull request merged**